sitestats/sitestats-impl-hib/src/java/org/sakaiproject/sitestats/impl/SitePresenceTotalImpl.java

```diff
@@ -50,7 +50,7 @@ public SitePresenceTotalImpl(SitePresence sp) {
         siteId = sp.getSiteId();
         userId = sp.getUserId();
         totalVisits = 1;
-        lastVisitTime = sp.getLastVisitStartTime();
+        lastVisitTime = sp.getDate();
```

Comment:
Why not try last visit start time first, then use the date if it is null?
lastVisitStartTime = sp.getLastVisitStartTime() != null ? sp.getLastVisitStartTime() : sp.getDate();

I have a question too. Why is the last visit start time ever null if there is a site presence record? This seems like some bad code is involved in writing those records in the first place.

Reply:
I don't know why but it is null, the column last visit start time is defined as "default null" so... it happens a lot now :(
We only recover the original line in [SAK-31769](https://sakaiproject.atlassian.net/browse/SAK-31760), but i guess your line will work too.

Reply:
It seems worth trying the last visit time stamp, just in case it is set. It is a bit weird that it is default null though. You'd expect that it would always be set, something like that. But we can look at that code smell in another ticket :)

Reply:
Maybe @buckett could help us...

We have some doubts about if is better

`
lastVisitStartTime = sp.getLastVisitStartTime() != null ? sp.getLastVisitStartTime() : sp.getDate();
`

than

`
lastVisitTime = sp.getDate();
`

Reply:
Agree with change the line after some checking.